fibsem/milling/patterning/patterns2.py

```diff
@@ -96,17 +110,118 @@ def to_dict(self):
             "height": self.height,
             "depth": self.depth,
             "rotation": self.rotation,
-            "path": self.path
+            "time": self.time,
+            "passes": self.passes,
+            "scan_direction": self.scan_direction,
+            "path": self.path,
+            "bitmap": self.bitmap,
+
         }
-    
+
     @classmethod
     def from_dict(cls, ddict: dict) -> "BitmapPattern":
         return cls(
             width=ddict["width"],
             height=ddict["height"],
             depth=ddict["depth"],
             rotation=ddict.get("rotation", 0),
-            path=ddict.get("path", ""),
+            time=ddict.get("time", 0),
+            passes=ddict.get("passes", 0),
+            scan_direction=ddict.get("scan_direction", "TopToBottom"),
+            path=ddict.get("path"),
+            bitmap=ddict.get("bitmap"),
+            point=Point.from_dict(ddict.get("point", DEFAULT_POINT_DDICT)),
+        )
+
+
+@dataclass
+class TrenchBitmapPattern(BasePattern):
+    width: float
+    depth: float
+    spacing: float
+    upper_trench_height: float
+    lower_trench_height: float
+    time: float = 0
+    point: Point = Point()
+    name: str = "TrenchBitmap"
+    flip_lower_pattern: bool = True
+    path: Optional[Union[str, PathLike]] = None
+    bitmap: Optional[NDArray[Any]] = None
+    shapes: Optional[List[FibsemPatternSettings]] = None
+    _advanced_attributes = ["time",]
+
+    def define(
+        self, protocol: dict, point: Point = Point()
+    ) -> list[FibsemBitmapSettings]:
+        point = self.point
+        width = self.width
+        spacing = self.spacing
+        upper_trench_height = self.upper_trench_height
+        lower_trench_height = self.lower_trench_height
+        depth = self.depth
+        time = self.time
+        path = self.path
+        bitmap = self.bitmap
+
+        # calculate the centre of the upper and lower trench
+        centre_lower_y = point.y - (spacing / 2 + lower_trench_height / 2)
+        centre_upper_y = point.y + (spacing / 2 + upper_trench_height / 2)
+
+        # mill settings
+        lower_pattern_settings = FibsemBitmapSettings(
+            width=width,
+            height=lower_trench_height,
+            depth=depth,
+            # Allows flipping of lower pattern without first loading the bitmap
+            rotation=np.pi if self.flip_lower_pattern else 0,
+            centre_x=point.x,
+            centre_y=centre_lower_y,
+            # TODO: check if rotation considered before or after scan direction
```

Comment:
Scan direction is based on pattern rotation of 0. E.g. If you use TopToBottom and rotate pattern 180 deg, it will effectively be BottomToTop